pkg/kube/cluster-init.sh

```diff
@@ -300,20 +300,30 @@ check_start_containerd() {
                 logmsg "Started k3s-containerd at pid:$containerd_pid"
         fi
         if [ -f /etc/external-boot-image.tar ]; then
+                # Is containerd up?
+                if ! /var/lib/k3s/bin/k3s ctr -a /run/containerd-user/containerd.sock info > /dev/null 2>&1; then
+                        logmsg "k3s-containerd not yet running for image import"
+                        return
+                fi
+
                 # NOTE: https://kubevirt.io/user-guide/virtual_machines/boot_from_external_source/
                 # Install external-boot-image image to our eve user containerd registry.
                 # This image contains just kernel and initrd to bootstrap a container image as a VM.
                 # This is very similar to what we do on kvm based eve to start container as a VM.
                 logmsg "Trying to install new external-boot-image"
                 # This import happens once per reboot
-                if ctr -a /run/containerd-user/containerd.sock image import /etc/external-boot-image.tar; then
-                        eve_external_boot_img_tag=$(cat /run/eve-release)
-                        eve_external_boot_img=docker.io/lfedge/eve-external-boot-image:"$eve_external_boot_img_tag"
-                        import_tag=$(tar -xOf /etc/external-boot-image.tar manifest.json | jq -r '.[0].RepoTags[0]')
-                        ctr -a /run/containerd-user/containerd.sock image tag "$import_tag" "$eve_external_boot_img"
-
-                        logmsg "Successfully installed external-boot-image $import_tag as $eve_external_boot_img"
-                        rm -f /etc/external-boot-image.tar
+                import_name_tag=$(tar -xOf /etc/external-boot-image.tar manifest.json | jq -r '.[0].RepoTags[0]')
+                import_name=$(echo "$import_name_tag" | cut -d ':' -f 1)
+                eve_external_boot_img_name="docker.io/lfedge/eve-external-boot-image"
+                if [ "$import_name" = "$eve_external_boot_img_name" ]; then
+                        if /var/lib/k3s/bin/k3s ctr -a /run/containerd-user/containerd.sock image import /etc/external-boot-image.tar; then
+                                eve_external_boot_img_tag=$(cat /run/eve-release)
+                                eve_external_boot_img="${eve_external_boot_img_name}:${eve_external_boot_img_tag}"
+                                if /var/lib/k3s/bin/k3s ctr -a /run/containerd-user/containerd.sock image tag "$import_name_tag" "$eve_external_boot_img"; then
```

Comment:
Is this also a safety check? If so, what happens if it fails?

Reply:
Yes another safety check, this will be attempted on the next run of check_start_containerd until the tar file is deleted on line 324

Reply:
yes

Reply:
That can get a bit expensive. Theoretically, containerd should recognize that it is a digest that it already has and not process it, but I don't even know if that actually is true, and if it is, at what stage it stops. Loading the blob is pretty cheap; checking the digest is more expensive.

Would it be easier to do an internal loop on the load? If it is successful, remove the tar file, so future runs only deal with the tag? Or conversely, before loading, check if the tag exists?

Reply:
It's only loaded once per boot event.  Right now it checks if the tar exists, does the import and then deletes the tar.  

On a system I have running currently the external-boot-image has been imported/tagged by many eve-os upgrades but all reference the same image id since it hasn't changed at all recently.

Reply:
@deitch would you prefer checking through ctr or crictl for the tag instead of the `if [ -f /etc/external-boot-image.tar ]; then`?

Reply:
Yeah, that probably makes more sense. It is like the other flags. Why use a flag, when it is low-cost to check if it exists already (or to import it, but that is less the case here, than it is with k8s manifests).

Is there concern that it might have changed? If so, then you have to import it. Other than that, checking if it exists makes the most sense.

Reply:
Oh yeah, that is clean to understand. Looks good.

Reply:
Thanks